### DISCUSSION TITLE
Reuse or Deactivate delete /users/{user_id} and how to protect resources

### DISCUSSION THREAD

**Discussion by migeruj** [fastapi-users/fastapi-users#606] [Q&A] [answered]
Hi guys I am experimenting with fastapi-users. But I have some questions about the user's endpoint. 

1. In my case I need to not be able to delete users, just update the is_active field instead of delete record. 
2. How can I protect the endpoints after they are logged in? I don't see clear documentation on this second point

I think that's all!

Thanks in advance.

**Comment by frankie567** [comment, marked as answer]
1. There is currently no easy way to remove the DELETE endpoint. Since it's only allowed for superusers, maybe you can keep it and just not use it. If you really want to remove it, you can loop through the routes of the router and remove the one with method delete. Something like this:

```py
users_router = fastapi_users.get_users_router()
users_router.routes = [route for route in users_router.routes if route.name != "delete_user"]
app.include_router(users_router, prefix="/users", tags=["users"])
```

2. The library provides a dependency for this. See the docs here: https://frankie567.github.io/fastapi-users/usage/dependency-callables/

**Reply by migeruj** [reply]
Terrific! Actually, I didn't pay attention to dependency callables. Thank you for providing the requested information. I am learning so much about fastapi now.

**Comment by migeruj** [comment]
Forgot to close. Thanks

**Comment by NickNaskida** [comment]
Note: `fastapi-users v1.12 ` route name changed to `users:delete_user`

```py
users_router.routes = [route for route in users_router.routes if route.name != "users:delete_user"]
```
